### PR TITLE
[Train] Fix the logging, llama-2 tokenizer issue, and s3 storage path metric recording

### PR DESCRIPTION
**Pull request by kouroshHakha**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/templates/04_finetuning_llms_with_deepspeed/finetune_hf_llm.py [hunk 1]
```diff
@@ -122,6 +123,16 @@ def evaluate(
         perplexity = float("inf")
     return perplexity, eval_loss
 
+def _test_tokenizer(model_name):
```
Comment (ArturNiederfahrenhorst) [inline doc/source/templates/04_finetuning_llms_with_deepspeed/finetune_hf_llm.py:126]: Maybe also leave a breadcrumb here for https://github.com/huggingface/transformers/issues/25176 ? Otherwise context for this test will be lost over time

File: doc/source/templates/04_finetuning_llms_with_deepspeed/finetune_hf_llm.py [hunk 2]
```diff
@@ -310,19 +324,24 @@ def training_function(kwargs: dict):
...
             # as long as this is not the last step report here
             if step != (train_ds_len // batch_size - 1):
+                aggregated_loss = torch.mean(accelerator.gather(loss[None])).item()
                 session.report(
                     {
                         "epoch": epoch,
                         "iteration": step,
-                        "train_loss": loss_sum.item() / (step + 1),
+                        "train_loss_batch": aggregated_loss,
+                        "avg_train_loss_epoch": None,
```
Comment (ArturNiederfahrenhorst) [inline doc/source/templates/04_finetuning_llms_with_deepspeed/finetune_hf_llm.py:333]: Maybe unify names to `x_per_epoch` or `x_epoch`?

File: doc/source/templates/04_finetuning_llms_with_deepspeed/finetune_hf_llm.py [hunk 3]
```diff
@@ -540,9 +564,11 @@ def main():
     # json file
     with open(args.special_token_path, "r") as json_file:
         special_tokens = json.load(json_file)["tokens"]
-
+    
+    artifact_storage = os.environ['ANYSCALE_ARTIFACT_STORAGE']
+    user_name = os.environ['ANYSCALE_USERNAME'] 
     storage_path = (
-        f"{os.environ['ANYSCALE_ARTIFACT_STORAGE']}/finetuning_llms_with_deepspeed"
+        f"{artifact_storage}/{user_name}/ft_llms_with_deepspeed/{args.model_name}"
```
Comment (ArturNiederfahrenhorst) [inline doc/source/templates/04_finetuning_llms_with_deepspeed/finetune_hf_llm.py:571]: Noice.

**Review by matthewdeng — APPROVED** [review]
